File: mobile/app/src/main/java/ua/nure/cleaningservice/util/InternetConnection.java
package ua.nure.cleaningservice.util;

import android.content.Context;
import android.net.ConnectivityManager;
import android.os.Build;

import androidx.annotation.NonNull;

public class InternetConnection {

    public static boolean checkConnection(@NonNull Context context) {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
            return ((ConnectivityManager) context.getSystemService
                    (Context.CONNECTIVITY_SERVICE)).getActiveNetwork() != null;
        } else {
            return true;
        }
    }
}

File: mobile/app/src/main/java/ua/nure/cleaningservice/util/InternetConnection.kt
package ua.nure.cleaningservice.util

import android.content.Context
import android.net.ConnectivityManager
import android.os.Build

object InternetConnection {
    fun checkConnection(context: Context): Boolean {
        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
            (context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager).activeNetwork != null
        } else {
            true
        }
    }
}
